crates/typst/src/image.rs

```diff
@@ -342,23 +342,50 @@ fn load_svg_fonts(
         name
     };
 
-    // Load fallback family.
-    let mut fallback_usvg_compatible = None;
-    if let Some(family) = loader.fallback_family() {
-        fallback_usvg_compatible = load(family);
-    }
-
-    // Find out which font families are referenced by the SVG.
+    // Determine the best font for each text node.
     traverse_svg(&tree.root, &mut |node| {
         let usvg::NodeKind::Text(text) = &mut *node.borrow_mut() else { return };
         for chunk in &mut text.chunks {
             for span in &mut chunk.spans {
-                for family in &mut span.font.families {
-                    if family.is_empty() || load(family).is_none() {
-                        if let Some(fallback) = &fallback_usvg_compatible {
-                            *family = fallback.into();
+                let text = chunk
+                    .text
+                    .chars()
+                    .skip(span.start)
+                    .take(span.end - span.start)
+                    .collect::<EcoString>();
+
+                // Find a font that covers all characters in the span while
+                // taking the fallback order into account.
+                let font = span
+                    .font
+                    .families
+                    .iter()
+                    .chain(loader.fallback_families().iter())
+                    .filter(|family| !family.is_empty())
+                    .find_map(|family| {
+                        let fonts = loader.load(family.to_lowercase().as_str());
```

Comment:
calling `loader.load` very often might put a lot of pressure on the incremental computation system. some sort of caching would be good, ideally shared for this and `load`.